### ISSUE TITLE
[Request] Add some kind of visual cue that HP has not froze and is working on the watch folders

### ISSUE THREAD

**Issue by HASJ** [Pewpews/happypanda#38]
Clear your library, set up a watch folder and drop 100 galleries in there.

It will look like HP is doing nothing at all. After checking the log, I discovered that HP was indeed monitoring the files, it was just going to finish ALL of them before giving me input on it. Adding a simple icon/text that HP has detected new files in the folder and is parsing them should mitigate this.

Also noticed that HP creates each files thumbnails after it parses all files. I think you could make it show the thumbs right after it goes through a file, akin to ComicRack. This would completely erase the feeling that HP is doing nothing when, in fact, it is monitoring the folder and working normally.

**Comment by twiddli** [comment]
I'll add an indicator _(like the one when fetching metadata)_, when scanning for new galleries.

> I think you could make it show the thumbs right after it goes through a file, akin to ComicRack. This would completely erase the feeling that HP is doing nothing when, in fact, it is monitoring the folder and working normally.

I'm pretty sure that's what it's already doing. Though with a small delay, which I'm going to decrease by the way. Or have i misunderstood?

**Comment by HASJ** [comment]
> > I think you could make it show the thumbs right after it goes through a file, akin to ComicRack. This would completely erase the feeling that HP is doing nothing when, in fact, it is monitoring the folder and working normally.
> 
> I'm pretty sure that's what it's already doing. Though with a small delay, which I'm going to decrease by the way. Or have i misunderstood?

Well, then it is not doing it for me. Giving a more detailed example, whenever ComicRack detects a file, it'll instantly add it's entry to the library and as soon as that file's thumb is ready, it'll fade-in to the gallery.

What I think you meant is that HP is creating the thumbs and yes, it is doing that after it goes through all the queued files. ComicRack creates the thumb and displays it right away in the GUI after going through a file and moves on to the next one very quickly.

I'm not worrying about speed right now, CR had 5+ years of development before getting to the point it is.

**Comment by HASJ** [comment]
A clip to illustrate what it is that ComicRack does: https://imgrush.com/gIQdIAJVw_bz.webm

It is much faster than this but extracting an archive, making a thumbnail, putting it back on CR and capturing the video is a bit much for my old warrior PC.

**Comment by twiddli** [comment]
Alright, I understand now. This is actually feasible but requires a great amount of work unfortunately. In the earlier versions I actually tried to make it do this.. _adding galleries dynamically_, but I scrapped the idea because of my inexperience and knowledge of the Qt framework at that time. Now it's a different story though. I'll add _dynamically adding galleries_ to the low priority section on my to-do list for now. I might try and implement this in a holiday.
